### ISSUE TITLE
Upgrade minimum acceptable cmake version

### ISSUE THREAD

**Issue by keithc-ca** [eclipse-openj9/openj9#19445]
cmake version 3.28.3 warns:
```
CMake Deprecation Warning at CMakeLists.txt:22 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

**Comment by keithc-ca** [comment]
Version 3.28.3 is the current default version from cygwin.

**Comment by pshipton** [comment]
@AdamBrousseau fyi

**Comment by keithc-ca** [comment]
It turns out we don't need to modify the minimum; see #22161.

**Comment by keithc-ca** [comment]
Reopening: https://github.com/eclipse-openj9/openj9/pull/22161 was reverted.

**Comment by AdamBrousseau** [comment]
Do we know why AIX didn't work? Cmake version I saw on the external machine was 3.17.1.

https://openj9-jenkins.osuosl.org/job/Build_JDK8_ppc64_aix_Nightly/1062/consoleFull
```
21:02:09  checking for cmake... /opt/cmake-3.17.1/bin/cmake
...
21:07:05  -- Starting with CMake version 3.17.1
...
21:07:29  Assembler:
21:07:29  /home/jenkins/workspace/Build_JDK8_ppc64_aix_Nightly/openj9/runtime/libffi/powerpc/aix.S: line 1: Error In Syntax 
21:07:29  /home/jenkins/workspace/Build_JDK8_ppc64_aix_Nightly/openj9/runtime/libffi/powerpc/aix.S: line 2: invalid opcode or pseudo-op
```

**Comment by pshipton** [comment]
I would have suspected some other change but there was nothing else that changed. My guess is that the change modified some cmake policy which caused the interpretation of the .S file to change on AIX.

**Comment by keithc-ca** [comment]
I haven't had time yet to dig into the reason for that. My expectation (like Peter's) is that there's some cmake policy change between 3.4 and 3.28 that affects how `.S` files are handled.

**Comment by AdamBrousseau** [comment]
Is this resolved then?
- https://github.com/eclipse-openj9/openj9/pull/22161
- https://github.com/eclipse-omr/omr/pull/7817
- https://github.com/eclipse-openj9/openj9/pull/22202
- https://github.com/eclipse-openj9/openj9/pull/22207

**Comment by keithc-ca** [comment]
Yes, I think it's time to close this. Going forward it will be less about adjusting the minimum version that can be used and more about adapting to policy changes that come with newer versions.
